pkg/controllers/managementuser/snapshotbackpopulate/snapshotbackpopulate.go

```diff
@@ -104,9 +106,13 @@ func (h *handler) configMapToSnapshots(configMap *corev1.ConfigMap) (result []rk
 		}
 		snapshot := rkev1.ETCDSnapshot{
 			Name:      file.Name,
+			Location:  file.Location,
```

Comment:
I think the reason the `snapshotFile` is duplicated is due to the fact that it represents a 1:1 replica of the snapshotFile that is used by K3s/RKE2 when they create the config map for unmarshaling purposes.

I think I might tackle refactoring this in a future PR, rather than this one.